src/format-error.ts

```diff
@@ -11,7 +11,7 @@ export class FormatError {
 
 	constructor(message: string, invalidValue: any) {
 		if (FormatError.ConditionType === null) {
-			FormatError.ConditionType = new ErrorConditionType("FormatError", "The value is not properly formatted.");
+			FormatError.ConditionType = new ErrorConditionType("FormatError", "The value is not properly formatted.", this);
```

Comment:
Passing `this` to the `ErrorConditionType` constructor here may not be correct, given that it appears to be some type of static initialization. @mattheyan are you familiar with this at all? In my view, "correctness" would hinge on how the condition type's `source` ends up being used. I.e. if downstream we reference something that would be the same for every format error, it would be fine, though that is not guaranteed by any means.

Reply:
Yeah, I think `FormatError` is basically just a wrapper around the condition. Seems like the more appropriate source would be the `Format` object.

Reply:
I think I was getting more at the fact we are providing a potentially random instance to a static object. Is that not a problem?

Reply:
Oh, I see. The condition type on format error is static. I missed that detail. Yeah, I agree.

Reply:
Maybe the solution is to not pass in the source to condition type (allow it to be null if that's not possible already), but pass in the `Format` instead in `createCondition` below?

Reply:
Of course, if we did that we'd have to check the condition's source, not the condition type.